lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -0,0 +1,129 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Task::Git::Clone;
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
+
+use OpenQA::Utils qw(run_cmd_with_log_return_error);
+use Cwd 'abs_path';
+
+sub register ($self, $app, @) {
+    $app->minion->add_task(git_clone => \&_git_clone);
+}
+
+sub _git_clone ($job, $clones) {
+    # $clones is a hashref with paths as keys and urls to git repos as values.
+    # The urls may also refer to a branch via the url fragment.
+   # If no branch is set, the default branch of the remote (if target path doesn't exist yet) or local checkout is used.
+# If the target path already exists, an update of the specified branch is performed, if the url matches the origin remote
+
+    my $app = $job->app;
+    my $job_id = $job->id;
+
+    # Prevent multiple git clone tasks for the same path to run in parallel
+    my @guards;
+    for my $path (sort keys %$clones) {
+        while (-l $path) { $path = abs_path($path); }    # guard symlink target
```

Comment:
This might loop endlessly on cyclic links. I suppose this should use [realpath](https://docs.mojolicious.org/Mojo/File#realpath).

Reply:
`realpath` doesn't resolve symlinks.